### ISSUE TITLE
[BUG] <title>最大输入长度--model_max_length参数传递无效

### ISSUE THREAD

**Issue by GenuineWWD** [OpenBMB/MiniCPM-V#545]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [X] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 当前行为 | Current Behavior

在训练脚本中，修改--model_max_length参数改变最大输入长度无效，依然会有warning。例如设置--model_max_length=4096，仍旧有The input length (3773) exceeds the model's maximum length (2048), so it has been truncated
No tokens available to compute loss.

### 期望行为 | Expected Behavior

修复bug，使得--model_max_length参数有效

### 运行环境 | Environment

```Markdown
- OS:Ubuntu 20.04
- Python:3.10
- Transformers:4.44
- PyTorch:2.1
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`):11.8
```

**Comment by LDLINGLINGLING** [comment]
你好，请尝试
![image](https://github.com/user-attachments/assets/49696cf8-fe13-4f54-8bce-a5e442117277)
在dataload中修改最大长度

**Comment by GenuineWWD** [comment]
> 你好，请尝试 ![image](https://private-user-images.githubusercontent.com/47373076/363548299-0a67f953-c0b6-4be7-86b6-5e78e5277ed9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUyNzA3NzYsIm5iZiI6MTcyNTI3MDQ3NiwicGF0aCI6Ii80NzM3MzA3Ni8zNjM1NDgyOTktMGE2N2Y5NTMtYzBiNi00YmU3LTg2YjYtNWU3OGU1Mjc3ZWQ5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTAyVDA5NDc1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM5ZTRkNTdiODM5YzExNWU0MDA5Y2RlZDY2NjE4ODNhMjI3OTYxMTQzMzRlNzFmYjAwMDQ0OWZmYThlNTM1NjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.guOp7l1-ht2vZ6MlmEc_uYtjVi3QxpyephjfgD-wsT4)
> 
> 在dataset.py中修改process函数中红框中最大长度

你好，尝试了也是不行，我也尝试了直接把preprocess函数的默认长度设成4096都不行。

**Comment by LDLINGLINGLING** [comment]
最近看了代码，warning应该不影响

**Comment by qyc-98** [comment]
可以直接把tokenizer的max length设置成你需要的长度
